includes/class-wcpdf-admin.php

```diff
@@ -782,44 +788,90 @@ public function regenerate_document() {
 			) );
 		}
 
-		$order_id = absint( $_POST['order_id'] );
-		$order = WCX::get_order( $order_id );
-		$document_type = sanitize_text_field( $_POST['document_type'] );
+		$order_id        = absint( $_POST['order_id'] );
+		$order           = WCX::get_order( $order_id );
+		$document_type   = sanitize_text_field( $_POST['document_type'] );
+		$action_type     = sanitize_text_field( $_POST['action_type'] );
+		$notice          = sanitize_text_field( $_POST['wpcdf_document_data_notice'] );
 
 		// parse form data
-		parse_str($_POST['form_data'], $form_data);
+		parse_str( $_POST['form_data'], $form_data );
 		if ( is_array( $form_data ) ) {
 			foreach ( $form_data as $key => &$value ) {
 				if ( is_array( $value ) && !empty( $value[$order_id] ) ) {
 					$value = $value[$order_id];
 				}
 			}
 		}
+		$form_data       = stripslashes_deep( $form_data );
+
+		// notice messages
+		$notice_messages = array(
+			'saved'       => array(
+				'success' => __( 'Document data saved!', 'woocommerce-pdf-invoices-packing-slips' ),
+				'error'   => __( 'An error ocurred while saving the document data!', 'woocommerce-pdf-invoices-packing-slips' ),
+			),
+			'regenerated' => array(
+				'success' => __( 'Document data regenerated!', 'woocommerce-pdf-invoices-packing-slips' ),
+				'error'   => __( 'An error ocurred while regenerating the document data!', 'woocommerce-pdf-invoices-packing-slips' ),
```

Comment:
It's not the data that is regenerated but the document, so you can leave out 'data'.